### ISSUE TITLE
Let question descriptions support embedded images

### ISSUE THREAD

**Issue by neel-a-mehta** [civiform/civiform#979]
To do this, we'd probably want to enable uploading of images through the question builder UI. The image could be added to the description through #888.

**Comment by mayamcgrory** [comment]
I would like to assign myself to this issue.

**Comment by elisekalstad** [comment]
This was a feature requested by a community feedback group on 2/9/2022.

**Comment by elisekalstad** [comment]
Idea: Could have a separate question type that is just "image".

**Comment by msprenke** [comment]
This may go hand in hand with the work needed for #3783

**Comment by rasmi** [comment]
Although this issue came first, #3783 is more active, so closing this issue as a duplicate.

**Comment by rasmi** [comment]
(re-opening because this is about program/question descriptions, not just the cards as described in #3783.

**Comment by caitlinshk** [comment]
Now that we've implemented a feature that lets admins upload images for program cards, we should be able to leverage that feature to add images for questions as well.

https://github.com/civiform/civiform/issues/5676 is the Epic issue for program card images. That issue links out to the PRD and TDD issues as well, so you can find the PRD and TDD there. All PRs that were submitted are also linked to #5676 if you want to take a look at all the PRs needed for the feature.

Two pieces of information that I want to call out about image upload:

- **Technical restriction**: The uploaded images are stored in a CE's cloud provider, not CiviForm's server. AWS's image upload has a restriction that **the image upload must be in a separate `<form>` element from anything else on the page**. This is because when the form is submitted with the image, the image is sent directly to the cloud provider, **not** to the CiviForm server. Once the image is successfully uploaded to the cloud provider, the cloud provider will redirect back to CiviForm with the file path, but it won’t contain any other information that was included in the `<form>`.  See [Restriction #2: AWS in the TDD](https://docs.google.com/document/d/1HU4gtZz8e4wgsR4xZ4tAkZlZd4bAd4fM8i5Ela9NoHI/edit#heading=h.qmw6qx4cx1a5).

  This means that we can’t update the question image at the same time as we update other things about the question like title or help text. The question image updates need to happen completely separately. There will need to be some UX work done to figure out where to put the question image upload page so that admins can find it easily, while keeping it separate from the rest of the question editing process.

- **Accessibility restriction**: Any image that we show to users must also have a text description so that users who are blind or visually impaired can understand what the image is showing. For program images, we required that the text description be set before the image is uploaded to meet this accessibility requirement.

### Implementation recommendations

Because images for questions have a lot of the same requirements as images for program cards, we can probably re-use a lot of the infrastructure. 

See [the file storage backend wiki page](https://github.com/civiform/civiform/wiki/File-Storage-Backend) for general information about how files are uploaded in CiviForm. We currently store files in two different buckets in cloud storage:
1. A private bucket that stores applicant-uploaded files (e.g. if an applicant needed to upload a photo of their driver's license)
2. A public bucket that stores files that all applicants need to see. Right now, the only files stored in the public bucket are the program images. I think we can re-use the same public bucket for question images so that deployments don’t need to set up more buckets with the same permissions.

Main steps:
- Update the `questions` table to include 2 new columns:
    1.  The path to the image in cloud storage (`summary_image_file_key` is what we use for program images)
    3. The image description for alt-text. (`localized_summary_image_description` is what we use for program images)
- Update `PublicFileNameFormatter` to provide a file key format for question images. I recommend something like `question-image/question-{questionId}/{originalFileName}` to make it similar to the program key.
- Create a controller similar to `AdminProgramImageController` and a view similar to `ProgramImageView`, but question equivalents. Some code may be able to be reused between the program image upload and question image upload.
- Display the question image and alt-text to applicants if the question has an image. We should also update the question preview that admins see as they're drafting a question.
- Create a durable job similar to `UnusedProgramImagesCleanupJob` that will remove any old question images that are no longer used.

**Comment by elisekalstad** [comment]
@caitlinshk -- Is this done and ready for deployment? (Seeing that there is a larger epic linked: https://github.com/civiform/civiform/issues/5676)

**Comment by caitlinshk** [comment]
Oops sorry, this is not done -- I mentioned the program card images epic because question images will be easier to implement now that we've done the program images work. This is ready for development, but it hasn't been started.

**Comment by msprenke** [comment]
@swatkat1  is this the Github item being used for images in questions or is there another one?

**Comment by ekalstad** [comment]
Yes! Here are links to PRD and TDD: [PRD](https://docs.google.com/document/d/185uXqVEvyk7kyyho3AQwfOxsE53fPYdEiQY11xJWZhg/edit?usp=sharing); [TDD](https://docs.google.com/document/d/1uG-2TKiFAERGpVRMMcU9OXGyM3hVsOVbR-ta3_rFP1I/edit?usp=sharing);

**Comment by myfriendgu** [comment]
Designs can be found here: https://www.figma.com/design/RQBLTI1YeeBlqyu4fsKHNk/Images-in-questions?node-id=1-19&p=f&t=BrtNSG1jyLiopvho-0

**Comment by swatkat1** [comment]
This Issue is in progress. The tracking epic is- https://github.com/civiform/civiform/issues/13786
